tests/Common/InjectorTest.php

```diff
@@ -67,6 +72,44 @@ public function testInvokeStatic(): void
         $this->assertIsBool($result);
     }
 
+    public function testInvokeCallStatic(): void
+    {
+        $container = $this->getContainer();
+
+        $result = (new Injector($container))->invoke([CallStaticObject::class, 'foo']);
+
+        $this->assertSame('bar', $result);
+    }
+
+    public function dataInvokeStaticWithStaticCalls(): array
+    {
+        return [
+            [CallStaticWithStaticObject::class],
+            [CallStaticWithSelfObject::class],
+            [StaticWithStaticObject::class],
+            [StaticWithSelfObject::class],
+        ];
+    }
+
+    /**
+     * @dataProvider dataInvokeStaticWithStaticCalls
+     */
+    public function testInvokeStaticWithStaticCalls(string $className): void
+    {
+        if (
+            $className === CallStaticWithStaticObject::class
+            && version_compare(PHP_VERSION, '8.1-dev', '<')
+        ) {
+            /** @link https://bugs.php.net/bug.php?id=81626 */
+            $this->markTestSkipped('Bug in PHP version below 8.1. See https://bugs.php.net/bug.php?id=81626');
+        }
```

Comment:
Todo: remove this

Reply:
@roxblnfk it will be lost here. Please create an issue.